cdap-cli/src/main/java/co/cask/cdap/cli/util/InstanceURIParser.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.cli.util;
+
+import co.cask.cdap.cli.CLIConfig;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.proto.Id;
+import com.google.inject.Inject;
+
+import java.net.URI;
+
+/**
+ */
+public class InstanceURIParser {
```

Comment:
Actually it requires `CConfiguration`, so putting it into `ConnectionInfo` isn't convenient.